File: practice/mathematics/fundamentals/is_fibo/IsFibo.js
'use strict';

let isFibo = n => {
    let fib0 = 0;
    let fib1 = 1;
    while(fib0 <= n) {
        if (fib0 == n) return "IsFibo";
        let temp = fib0;
        fib0 = fib1;
        fib1 += temp;
    }
    return "IsNotFibo"
};
const processData = input => {
    console.log(input.split("\n").slice(1).map(i => parseInt(i)).map(isFibo).join("\n"));
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/mathematics/fundamentals/is_fibo/is_fibo.rb
def is_fibo(number)
  fib0 = 0
  fib1 = 1
  while fib0 <= number do
    return "IsFibo" if fib0 == number
    temp = fib0
    fib0 = fib1
    fib1 += temp
  end
  return "IsNotFibo"  
end

test_cases = gets.chomp.to_i
test_cases.times do
  number = gets.chomp.to_i
  result = is_fibo(number)
  puts result
end